ntbs-service/Startup.cs

```diff
@@ -175,23 +175,26 @@ public void ConfigureServices(IServiceCollection services)
             });
             SetupHangfire(services);
 
-            // DB Contexts
-            services.AddDbContext<NtbsContext>(options =>
-                options.UseSqlServer(Configuration.GetConnectionString("ntbsContext"))
-            );
+            var auditDbConnectionString = Configuration.GetConnectionString("auditContext");
 
-            services.AddSingleton<NtbsContextDesignTimeFactory>();
+            if (!Env.IsEnvironment("CI")) // Maybe we should check if connection strings are null instead?
+            {
+                // DB Contexts
+                services.AddDbContext<NtbsContext>(options =>
+                    options.UseSqlServer(Configuration.GetConnectionString("ntbsContext"))
+                );
 
-            var auditDbConnectionString = Configuration.GetConnectionString("auditContext");
+                services.AddSingleton<NtbsContextDesignTimeFactory>();
 
-            services.AddDbContext<AuditDatabaseContext>(options =>
-                options.UseSqlServer(auditDbConnectionString)
-            );
+                services.AddDbContext<AuditDatabaseContext>(options =>
+                    options.UseSqlServer(auditDbConnectionString)
+                );
 
-            // Add a DbContext for Data Protection key storage
-            services.AddDbContext<KeysContext>(options =>
-                options.UseSqlServer(Configuration.GetConnectionString("keysContext")));
-            services.AddDataProtection().PersistKeysToDbContext<KeysContext>();
+                // Add a DbContext for Data Protection key storage
+                services.AddDbContext<KeysContext>(options =>
+                    options.UseSqlServer(Configuration.GetConnectionString("keysContext")));
+                services.AddDataProtection().PersistKeysToDbContext<KeysContext>();
+            }
```

Comment:
After changing the host that we build the integration tests were not happy running with null connection strings for the DBs - I'm not 100% sure why, before it just wasn't trying to connect to a SQL database which is what we wanted. It seems after the change it now won't accept any null string as a connection string. Thoughts?

Reply:
It seems to be a difference in behaviour of the two types of host.
On master: we build a web host. This call to "AddDbContext" runs as we configure it, but then when we `.Build()` the web host we **do not** call `UseSqlServer(..)`, which results in no error (I'm guessing because it knows the connection string is null and so should not try setup the service?
On this branch: we build a generic host. This call to "AddDbContext" runs as we configure it, when we `.Build()` the host we call `UseSqlServer(..)`, which results in an error.